db/hints/host_filter.hh

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <functional>
+#include <unordered_set>
+#include <exception>
+#include <iostream>
+#include <string_view>
+
+#include <seastar/core/sstring.hh>
+#include "gms/inet_address.hh"
+#include "locator/snitch_base.hh"
+#include "seastarx.hh"
+
+namespace db {
+namespace hints {
+
+// host_filter tells hints_manager towards which endpoints it is allowed to generate hints.
+class host_filter final {
+private:
+    enum class enabled_kind {
+        enabled_for_all,
+        enabled_selectively,
+        disabled_for_all,
+    };
+
+    enabled_kind _enabled_kind;
+    std::unordered_set<sstring> _dcs;
+
+    static std::string_view enabled_kind_to_string(host_filter::enabled_kind ek);
+
+public:
+    struct enabled_for_all_tag {};
+    struct disabled_for_all_tag {};
+
+    // Creates a filter that does not allow any hints (default)
```

Comment:
I think it makes more sense to have the same default as in the application, which is `enabled to all hosts`.

Reply:
I agree. Changed.